### PR TITLE
Changed Material Dialog with Alert Dialog for DeckPicker.kt

### PR DESCRIPTION
**Pull request by PalashKohli007**
## Purpose / Description
_Describe the problem or feature and motivation_

## Fixes
Fixes _Link to the issues._

## Approach
_How does this change address the problem?_

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration (SDK version(s), emulator or physical, etc)

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 1]
```diff
@@ -1729,9 +1728,9 @@ open class DeckPicker :
                             return@setOnKeyListener true
                         }
                         // Show confirmation dialog to check if the user wants to cancel the sync
-                        MaterialDialog(mProgressDialog!!.context).show {
+                        AlertDialog.Builder(mProgressDialog!!.context).show {
                             message(R.string.cancel_sync_confirm)
-                            cancelable(false)
+                                .setCancelable(false)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1733]: `.cancelable(false)` is available in `main`

**Review by david-allison — CHANGES_REQUESTED** [review]
LGTM, one nitpick

**Comment by david-allison** [comment]
BTW, could you `rebase` rather than opening new PRs for review changes, this causes comments/review history to be lost

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 2]
```diff
@@ -2910,10 +2909,10 @@ open class DeckPicker :
         }
         // TODO: maybe handle onStorageMigrationCompleted()
         // TODO: sync_impossible_during_migration needs changing
-        MaterialDialog(this).show {
+        AlertDialog.Builder(this).show {
```
Comment (criticalAY) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:2912]: This change is not actually worth it as the pending PR will replace this Dialog as we will be having a DialogFragment so I suggest reverting these lines inside `warnNoSyncDuringMigration`

**Review by criticalAY — CHANGES_REQUESTED** [review]

**Comment by criticalAY** [comment]
@PalashKohli007 for future reference please fill out the PR template properly and make sure you fill out the test section and if possible attach screenshot it making any UI change

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 3]
```diff
@@ -2910,10 +2909,10 @@ open class DeckPicker :
...
             message(text = resources.getString(R.string.sync_impossible_during_migration, 5) + text)
-            positiveButton(res = R.string.dialog_ok)
-            negativeButton(res = R.string.scoped_storage_learn_more) {
+            positiveButton(R.string.dialog_ok)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:2914]: refactor the library to use `res` instead of `stringRes so there's no changes here

**Review by david-allison — CHANGES_REQUESTED** [review]

**Review by criticalAY — APPROVED** [review]
Looks good to me, Thanks!

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 4]
```diff
@@ -622,7 +622,7 @@ open class DeckPicker :
                 positiveButton(R.string.close) {
                     exit()
                 }
-                cancelable(false)
+                    .setCancelable(false)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:625]: change

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 5]
```diff
@@ -1295,8 +1295,7 @@ open class DeckPicker :
                     negativeButton(R.string.close) {
                         recreate()
                     }
-                    cancelOnTouchOutside(false)
-                    cancelable(false)
+                        .setCancelable(false)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1298]: fix indentation

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 6]
```diff
@@ -1729,9 +1728,9 @@ open class DeckPicker :
                             return@setOnKeyListener true
                         }
                         // Show confirmation dialog to check if the user wants to cancel the sync
-                        MaterialDialog(mProgressDialog!!.context).show {
+                        AlertDialog.Builder(mProgressDialog!!.context).show {
                             message(R.string.cancel_sync_confirm)
-                            cancelable(false)
+                                .cancelable(false)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1733]: fix indentation

**Review by david-allison — CHANGES_REQUESTED** [review]
Looks good, only nitpicks

**Comment by PalashKohli007** [comment]
@david-allison , resolved comments , please review

**Review by david-allison — APPROVED** [review]
Awesome, thanks!

**Comment by PalashKohli007** [comment]
> Awesome, thanks!

welcome David

**Review by mikehardy — APPROVED** [review]
Nice!

**Pull request merged**